examples/dataflow-flex-python/README.md

```diff
@@ -0,0 +1,247 @@
+--disable-public-ips \
+--network projects/$HOST_PROJECT_ID/global/networks/$NETWORK \
+--subnetwork https://www.googleapis.com/compute/v1/projects/$HOST_PROJECT_ID/regions/$LOCATION/subnetworks/$SUBNET \
+--service-account-email dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com \
+--staging-location gs://$STAGING_BUCKET_NAME/staging \
+--temp-location gs://$STAGING_BUCKET_NAME/tmp \
+--region $LOCATION --worker-region=$LOCATION \
+--parameters output=$PROJECT_ID:$BQ_DATASET \
+--parameters input=gs://$INPUT_BUCKET_NAME/data/* \
+--parameters dead_letter_dir=gs://$INPUT_BUCKET_NAME/invalid_files \
+--parameters sdk_location=container \
+--parameters sdk_container_image=$LOCATION-docker.pkg.dev/$PROJECT_ID/$REPO/dataflow-2.40-base:dev \
+--project $PROJECT_ID
+```
+
+### CI/CD using Cloudbuild
+
+#### Build Docker Image and Template with Cloud build
+The below section uses gcloud command. In Real World scenario Cloud build triggers
+can be created ahich can run this build job whenever their is change in the code.
+```
+
+# Build and push Base Image
+gcloud builds submit --config cloudbuild_base.yaml . --project $PROJECT_ID --substitutions _LOCATION=$LOCATION,_PROJECT_ID=$PROJECT_ID,_REPOSITORY=$REPO
+
+# Build and push launcher image and create flex template
+gcloud builds submit --config cloudbuild_df_job.yaml . --project $PROJECT_ID --substitutions _LOCATION=$LOCATION,_PROJECT_ID=$PROJECT_ID,_REPOSITORY=$REPO,_TEMPLATE_PATH=gs://$INPUT_BUCKET_NAME/dataflow-templates
+```
+
+### Run Dataflow Flex template job from Composer Dags
+
+#### Set Environment Variables for composer
+```
+export COMPOSER_ENV_NAME=<composer-env-name>
+export COMPOSER_REGION=$LOCATION
+
+
+COMPOSER_VAR_FILE=composer_variables.json
+if [ ! -f "${COMPOSER_VAR_FILE}" ]; then
+    envsubst < composer_variables.template > ${COMPOSER_VAR_FILE}
+fi
+
+gcloud composer environments storage data import \
+--environment ${COMPOSER_ENV_NAME} \
+--location ${COMPOSER_REGION} \
+--source ${COMPOSER_VAR_FILE}
+
+gcloud composer environments run \
+${COMPOSER_ENV_NAME} \
+--location ${COMPOSER_REGION} \
+variables import -- /home/airflow/gcs/data/${COMPOSER_VAR_FILE}
+```
+
+#### Assign permissions to Composer Worker SA
+```
+COMPOSER_SA=$(gcloud composer environments describe  $COMPOSER_ENV_NAME --location $COMPOSER_REGION --project $PROJECT_ID --format json | jq '.config.nodeConfig.serviceAccount')
+
+# Assign Service Account User permissions
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:$COMPOSER_SA" --role roles/iam.serviceAccountUser
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:$COMPOSER_SA" --role roles/dataflow.admin
+```
+
+#### Upload the dag to composer's dag bucket
+```
+DAG_PATH=$(gcloud composer environments describe $COMPOSER_ENV_NAME --location $COMPOSER_REGION --project $PROJECT_ID --format json | jq '.config.dagGcsPrefix')
+gcloud storage cp dag/xml-to-bq-dag.py $DAG_PATH
+```
+
```

Comment:
Please add a section on the limitations of this approach to reading XML files. Using the file.read() method loads the whole XML file into memory for the conversion to dict via xmltodict. This approach works for small files but is not parallelizable on super large XML files as they are not read in chunks but in one go. This risks having a single worker dealing with very large file instances (slow) and running potentially out of memmory.

I suggest adding an estimate on limits to the sizes (per file) to be processing using this method. In my experience any XML file above ~ 300mb would start slowing down the pipeline considerably and potentially memory failures can start showing up at ~ 500mb. This is if you go with the default worker.

Reply:
Added it to the https://github.com/GoogleCloudPlatform/professional-services/pull/957/files#diff-0c0e69440b1541eb232f8210cbc88b32803a6d6232063defe72a661ea9c1d554